src/sagemaker/serverless/model.py

```diff
@@ -0,0 +1,90 @@
+# Copyright 2017-2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+"""Models that can be deployed to serverless compute."""
+from __future__ import absolute_import
+
+import time
+from typing import Optional
+
+import boto3
+import botocore
+
+from sagemaker.model import ModelBase
+
+from .predictor import LambdaPredictor
+
+
+class LambdaModel(ModelBase):
+    """A model that can be deployed to Lambda."""
+
+    def __init__(
+        self, image_uri: str, role: str, client: Optional[botocore.client.BaseClient] = None
+    ) -> None:
+        """Initialize instance attributes.
+
+        Arguments:
+            image_uri: URI of a container image in the Amazon ECR registry. The image
+                should contain a handler that performs inference.
+            role: The Amazon Resource Name (ARN) of the IAM role that Lambda will assume
+                when it performs inference
+            client: The Lambda client used to interact with Lambda.
+        """
+        self._client = client or boto3.client("lambda")
+        self._image_uri = image_uri
+        self._role = role
+
+    def deploy(
+        self, function_name: str, timeout: int, memory_size: int, wait: bool = True
+    ) -> LambdaPredictor:
+        """Create a Lambda function using the image specified in the constructor.
+
+        Arguments:
+            function_name: The name of the function.
+            timeout: The number of seconds that the function can run for before being terminated.
+            memory_size: The amount of memory in MB that the function has access to.
+            wait: If true, wait until the deployment completes (default: True).
+
+        Returns:
+            A LambdaPredictor instance that performs inference using the specified image.
+        """
+        response = self._client.create_function(
+            FunctionName=function_name,
+            PackageType="Image",
+            Role=self._role,
+            Code={
+                "ImageUri": self._image_uri,
+            },
+            Timeout=timeout,
+            MemorySize=memory_size,
+        )
+
+        if not wait:
+            return LambdaPredictor(function_name, client=self._client)
+
+        # Poll function state.
+        polling_interval = 5
+        while response["State"] == "Pending":
```

Comment:
I disagree that moving the polling interval to a module-level constant would improve reusability and readability.
1. The polling interval is specific to the deploy method, so it would not be used elsewhere in the module. 
2. Variables should generally be defined close to where they're used; if we move the variable to the top of the module, it would be far away from where it's used, and that might hurt readability.
3. On a related note, if we define the polling interval at the top of the module, it's unclear how that constant is used until much later (unless we add a note or name it something verbose like `DEPLOYMENT_POLLING_INTERVAL`)

I don't feel as strongly about the Lambda state constants, but I'm thinking it might be better not to use module-level constants fom them too.

1. I don't think `response["State"] == LAMBDA_PENDING_STATE` is clearer than `response["State"] == "Pending"`, although I do feel the former is more verbose.
2. The name of Lambda states won't ever change (e.g., the active state will always be "Active"), so we don't need to concern ourselves with changing the value of the constant (i.e., maintainability isn't that big of a factor).

Reply:
👍  We can revisit this when/if the file grows bigger

Reply:
To add, Regards point (2) in the first section, yes agreed with defining variable names close to where they are used, however as this codebase will grow, it will become harder to find these values buried among the relevant sections in case we need to change it. Referring to [YAGNI](https://martinfowler.com/bliki/Yagni.html), I do think we can revisit this when/if the file grows bigger 👍